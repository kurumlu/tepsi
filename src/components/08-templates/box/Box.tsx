import * as React from 'react';
import { FunctionComponent, MouseEvent, Ref } from 'react';
import cn from 'classnames';

export enum BoxTheme {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
  SECONDARY_LIGHT = 'secondary-light',
  TERTIARY = 'tertiary',
  TERTIARY_LIGHT = 'tertiary-light',
  QUATERNARY = 'quaternary',
  EXTERNAL = 'external',
  WHITE = 'white',
  NOTICE = 'notice',
  NONE = 'none',
}

export type BoxOpacity = 0 | 0.25 | 0.5 | 0.75;

export type SpacingScaleMap =
  | 0
  | 0.25
  | 0.5
  | 1
  | 1.5
  | 2
  | 2.5
  | 3
  | 3.5
  | 4
  | 5
  | 6
  | 7
  | 8
  | 9
  | 10
  | 12
  | 16;

export type SpacingProps = {
  margin?: SpacingScaleMap;
  marginTop?: SpacingScaleMap;
  marginBottom?: SpacingScaleMap;
  marginLeft?: SpacingScaleMap;
  marginRight?: SpacingScaleMap;
  padding?: SpacingScaleMap;
  paddingTop?: SpacingScaleMap;
  paddingBottom?: SpacingScaleMap;
  paddingLeft?: SpacingScaleMap;
  paddingRight?: SpacingScaleMap;
};

type BoxViewportProps = {
  spacing?: SpacingProps;
  theme?: BoxTheme;
};

export type FlexObject = {
  shrink?: number;
  grow?: number;
  basis?: number | string;
};

type BoxFlexProps = number | string | FlexObject;

export type BoxProps = {
  analyticsAreaName?: string;
  className?: string;
  dataQA?: string;
  mobile?: BoxViewportProps;
  tablet?: BoxViewportProps;
  desktop?: BoxViewportProps;
  mobileBleed?: boolean;
  tabletBleed?: boolean;
  withShadow?: boolean;
  flex?: BoxFlexProps;
  hidden?: boolean;
  opacity?: BoxOpacity;
  interactive?: boolean;
  tabIndex?: number;
  role?: string;
  id?: string;
  innerRef?: Ref<HTMLDivElement>;
  onClick?: (e?: MouseEvent<HTMLElement>) => void;
};

export const dashed = (s: string) => s.replace(/[A-Z]/g, '-$&').toLowerCase();
export const dotToUnderscore = (s?: number) =>
  s?.toString().replace(/\./g, '_');

export const getCssProperties = (
  viewport?: BoxViewportProps,
  state = ''
): string => {
  if (!viewport || !(viewport.spacing || viewport.theme)) return '';

  const classNames: string[] = [];
  const entries = viewport?.spacing ? Object.entries(viewport.spacing) : [];
  const theme = viewport?.theme;

  if (entries.length > 0) {
    for (const space of entries) {
      const [property, value] = space;

      classNames.push(`${dashed(property)}-${state}${dotToUnderscore(value)}`);
    }
  }
  if (theme) {
    classNames.push(`as-t-box-${state}-${theme}`);
  }

  return classNames.join(' ');
};

const getFlexProperties = (flex?: BoxFlexProps) => {
  if (!flex) {
    return undefined;
  }

  if (typeof flex !== 'object') {
    return { flex };
  }

  return Object.keys(flex).reduce((properties, flexKey) => {
    const flexProperty = `flex${
      flexKey[0].toUpperCase() + flexKey.slice(1).toLowerCase()
    }`;
    return {
      ...properties,
      [flexProperty]: flex[flexKey],
    };
  }, {});
};

export const Box: FunctionComponent<BoxProps> = ({
  analyticsAreaName,
  children,
  className,
  dataQA,
  mobile,
  tablet,
  desktop,
  mobileBleed,
  tabletBleed,
  withShadow,
  flex,
  hidden,
  opacity,
  interactive,
  role,
  tabIndex,
  id,
  innerRef,
  onClick,
}) => (
  <div
    className={cn(
      'as-t-box',
      {
        [`as-t-box--opacity-${dotToUnderscore(opacity)}`]: opacity,
        [`as-t-box--mobile-bleed`]: mobileBleed,
        'as-t-box--tablet-bleed': tabletBleed,
        'as-t-box--with-shadow': withShadow,
        'as-t-box--hidden': hidden,
        'as-t-box--interactive': interactive,
      },
      getCssProperties(mobile, 'mobile-'),
      getCssProperties(tablet, 'tablet-'),
      getCssProperties(desktop, 'desktop-'),
      className
    )}
    style={getFlexProperties(flex)}
    data-qa={dataQA}
    data-analytics-bannerid={analyticsAreaName}
    role={role}
    tabIndex={tabIndex}
    id={id}
    ref={innerRef}
    onClick={onClick}>
    {children}
  </div>
);
