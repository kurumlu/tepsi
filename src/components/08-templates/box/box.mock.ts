export const mockText =
  'Thatâ€™s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.';
