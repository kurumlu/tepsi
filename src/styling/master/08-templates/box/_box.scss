$t-box--background-color: transparent !default;
$t-box--mobile-bleed--marginX: -(get-spacing(2)) !default;
$t-box--tablet-bleed--marginX: -(get-spacing(3)) !default;

$t-box--with-shadow--box-shadow: 0 get-spacing(0.5) get-spacing(2) $t2-color__fade-dark-15 !default;

$box-themes: (
  'primary': $t2-color__gray-10,
  'secondary': rgba(get-ui-color(g), 0.1),
  'secondary-light': rgba(get-ui-color(g, light), .15),
  'tertiary': rgba(get-ui-color(b), 0.1),
  'tertiary-light': rgba(get-ui-color(b, regular), .05),
  'quaternary': rgba(get-ui-color(r, light), .15),
  'external': $t-box--background-color,
  'white': $t2-color__white,
  'notice': rgba($t2-color__notice, 0.1),
  'none': $t-box--background-color,
) !default;

$box-opacity: 0, 0.25, 0.5, 0.75;

$t-box--theme--color: $t2-color__foreground !default;
$t-box--theme--line-height: $t2-line-height__s !default;
$t-box--theme--heading--font-map: $t2-font-map__heading !default;
$t-box--theme--heading--font-size: $t2-font-size__body-m !default;
$t-box--theme--heading--line-height: $t2-line-height__xxs !default;
$t-box--theme--heading--margin-bottom: get-spacing(2) !default;

.as-t-box {
  background-color: $t-box--background-color;
}

@each $key, $box-theme in $box-themes {
  @each $device in (mobile, tablet, desktop) {
    .as-t-box-#{$device}--#{$key} {
      @include mq($from: #{$device}) {
        background-color: $box-theme;
        color: $t-box--theme--color;
        line-height: $t-box--theme--line-height;
      }
    }
  }
}

.as-t-box--mobile-bleed {
  @include mq($until: tablet) {
    margin-left: $t-box--mobile-bleed--marginX;
    margin-right: $t-box--mobile-bleed--marginX;
  }
}

.as-t-box--tablet-bleed {
  @include mq($from: tablet, $until: desktop) {
    margin-left: $t-box--tablet-bleed--marginX;
    margin-right: $t-box--tablet-bleed--marginX;
  }
}

.as-t-box--with-shadow {
  box-shadow: $t-box--with-shadow--box-shadow;
}

@each $key in $box-opacity {
  .as-t-box--opacity-#{parse-dotless($key)} {
    opacity: $key;
  }
}

.as-t-box--hidden {
  display: none;
}

.as-t-box--interactive {
  cursor: pointer;
}
