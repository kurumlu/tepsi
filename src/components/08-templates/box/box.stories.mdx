import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { Box, BoxTheme, SpacingProps } from './Box';
import { mockText } from './box.mock';
import { Text } from '../../05-atoms/text/Text';


<Meta
  title="Design system/Patterns/Templates/Box"
  component={Box}
/>

# Box Component

> Box component wraps a content to a div to modify an elementâ€™s appearance.

---

## Available props:

<ArgsTable of={Box} />


### FlexObject

| Name      | Type                 |
|-----------|----------------------|
| `grow`    | *number*             |
| `shrink`  | *number*             |
| `basis`   | *number | string*    |

---

> Property `spacing` depends on viewport and takes a list of different margin/padding values.
Values are not any and should be used from sass spacing variables list `$spacing-scale-map`:

```javascript
$spacing-scale-map: (0, .25, .5, 1, 1.5, 2, 2.5, 3, 4, 5, 6, 7, 8, 9, 10, 12, 16);
```
### With margin

```javascript
<Box
  desktop={{ spacing: { marginLeft: 2, marginRight: 2 }}}
  mobile={{ spacing: { marginLeft: 1, marginRight: 1 }}}>
  <Text>{mockText}</Text>
</Box>
```

<Canvas>
  <Story name="With margin">
    <Box
      desktop={{ spacing: { marginLeft: 2, marginRight: 2 }}}
      mobile={{ spacing: { marginLeft: 1, marginRight: 1 }}}>
      <Text>{mockText}</Text>
    </Box>
  </Story>
</Canvas>

### With padding

```javascript
<Box
  desktop={{ spacing: { paddingLeft: 2, paddingRight: 2 }}}
  mobile={{ spacing: { paddingLeft: 1, paddingRight: 1 }}}>
  <Text>{mockText}</Text>
</Box>
```

<Canvas>
  <Story name="With padding">
    <Box
      desktop={{ spacing: { paddingLeft: 2, paddingRight: 2 }}}
      mobile={{ spacing: { paddingLeft: 1, paddingRight: 1 }}}>
      <Text>{mockText}</Text>
    </Box>
  </Story>
</Canvas>

### With margin and theme

```javascript
<Box
  desktop={{ spacing: { marginLeft: 2, marginRight: 2 }}}
  mobile={{ spacing: { marginLeft: 1, marginRight: 1 }, theme: BoxTheme.SECONDARY}}>
  <Text>{mockText}</Text>
</Box>
```

<Canvas>
  <Story name="With margin and theme">
    <Box
      desktop={{ spacing: { marginLeft: 2, marginRight: 2 }}}
      mobile={{ spacing: { marginLeft: 1, marginRight: 1 }, theme: BoxTheme.SECONDARY}}>
      <Text>{mockText}</Text>
    </Box>
  </Story>
</Canvas>

---

### With shadow

```javascript
<Box withShadow mobile={{ theme: BoxTheme.PRIMARY}}>
  <Text>{mockText}</Text>
</Box>
```

<Canvas>
  <Story name="With shadow">
    <Box
      withShadow
      mobile={{ spacing: { padding: 4, margin: 4 }, theme: BoxTheme.PRIMARY}}>
      <Text>{mockText}</Text>
      <Text>{mockText}</Text>
      <Text>{mockText}</Text>
      <Text>{mockText}</Text>
    </Box>
  </Story>
</Canvas>

### With flex properties

```javascript
<Box flex={{ grow: 1, shrink: 1 }}>
  <Text>{mockText}</Text>
</Box>
```

<Canvas>
  <Story name="With flex properties">
    <Box
      flex={{ grow: 1, shrink: 1 }}>
      <Text>{mockText}</Text>
    </Box>
  </Story>
</Canvas>

---

### With hidden

```javascript
<Box hidden>
  <Text>{mockText}</Text>
</Box>
```

<Canvas>
  <Story name="With hidden">
    <Box hidden>
      <Text>{mockText}</Text>
    </Box>
  </Story>
</Canvas>

---

## Themes

| BoxTheme          |
|-------------------|
| `PRIMARY`         |
| `SECONDARY`       |
| `SECONDARY_LIGHT` |
| `TERTIARY`        |
| `TERTIARY_LIGHT`  |
| `QUATERNARY`      |
| `EXTERNAL`        |
| `WHITE`           |
| `NOTICE`          |
| `NONE`            |

Themes are optional and alter the look of the box and its content. For example a background color, font size, etc...
Background Theme is integrated in viewPortProps such that it can specifically be configured by media type

### Without theme
<Canvas>
  <Story name="Without theme">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### Primary theme
<Canvas>
  <Story name="With primary theme">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.PRIMARY}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### Secondary theme
<Canvas>
  <Story name="With secondary theme">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.SECONDARY}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### Secondary light theme
<Canvas>
  <Story name="With secondary light theme">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.SECONDARY_LIGHT}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### Tertiary theme
<Canvas>
  <Story name="With tertiary theme">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.TERTIARY}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### Tertiary light theme
<Canvas>
  <Story name="With tertiary light theme">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.TERTIARY_LIGHT}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### Quaternary theme
<Canvas>
  <Story name="With quaternary theme">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.QUATERNARY}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### External theme

> Transparent background colour, but with theme typography features.
> Needs for Product Placement copy components.

<Canvas>
  <Story name="With external theme">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.EXTERNAL}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### White theme
<Canvas>
  <Story name="With white theme">
    <Box
      mobile={{ spacing: { padding: 3 }, theme: BoxTheme.QUATERNARY}}>
      <Box
        desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
        tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
        mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.WHITE}}>
        <Text>
          <b>WHITE Box inside QUATERNARY Box</b><br/>
          {mockText}
        </Text>
      </Box>
    </Box>
  </Story>
</Canvas>

### Notice theme
<Canvas>
  <Story name="With notice theme">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.NOTICE}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### Primary theme only for mobile devices
Theme `primary` is only visible on mobile devices

<Canvas>
  <Story name="With primary theme only for mobile devices">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }, theme: BoxTheme.NONE}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.PRIMARY}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

### Secondary theme only in desktop devices
Theme `secondary` is only visible only on desktop devices
<Canvas>
  <Story name="With secondary theme only in desktop devices">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }, theme: BoxTheme.SECONDARY}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

---

## Bleed

> In printing, bleed is printing that goes beyond the edge of where the sheet will be trimmed. In other words, the bleed is the area to be trimmed off. The bleed is the part on the side of a document that gives the printer a small amount of space to account for movement of the paper, and design inconsistencies.

The `mobileBleed` and `tabletBleed` props adds a negative margin equal to the main content padding. This adds a kind of "bleed" effect to the component and causes it to stretch across the full width of the page on mobile/tablet.

<Canvas>
  <Story name="With primary theme and bleed effect">
    <Box
      mobileBleed
      tabletBleed
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.PRIMARY}}>
      <Text>
        <b>Test title</b><br/>
        {mockText}
      </Text>
    </Box>
  </Story>
</Canvas>

---

## With opacity


```javascript
$box-opacity: 0.25, 0.5, 0.75;
```

<Canvas>
  <Story name="With opacity">
    <Box
      desktop={{ spacing: { padding: 3, marginBottom: 5 }}}
      tablet={{ spacing: { padding: 3, marginBottom: 4 }}}
      mobile={{ spacing: { padding: 2, marginBottom: 2 }, theme: BoxTheme.PRIMARY}}
      opacity={0.5}>
      <Text>{mockText}</Text>
    </Box>
  </Story>
</Canvas>
