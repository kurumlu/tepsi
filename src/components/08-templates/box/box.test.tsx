import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { mockText } from './box.mock';

import {
  Box,
  BoxProps,
  dashed,
  dotToUnderscore,
  getCssProperties,
} from './Box';

describe('<Box />', () => {
  const component: ShallowWrapper<BoxProps> = shallow(<Box>{mockText}</Box>);

  it('renders <Box /> component', () => {
    expect(component.prop('className')).toEqual('as-t-box');
    expect(component.text()).toEqual(mockText);
    expect(component).toMatchSnapshot();
  });

  it('renders <Box /> component with `desktop` and `mobile` properties', () => {
    const componentWithSpacing: ShallowWrapper<BoxProps> = shallow(
      <Box
        desktop={{ spacing: { marginLeft: 2 } }}
        mobile={{ spacing: { marginLeft: 1 } }}>
        {mockText}
      </Box>
    );

    expect(componentWithSpacing.prop('className')).toEqual(
      'as-t-box margin-left-mobile-1 margin-left-desktop-2'
    );
    expect(componentWithSpacing).toMatchSnapshot('with props');
  });

  describe('<Box /> component util methods', () => {
    it('cals `dashed` util, replace all letters in capital to dashes', () => {
      expect(dashed('marginX')).toEqual('margin-x');
      expect(dashed('marginXYZ')).toEqual('margin-x-y-z');
    });

    it('cals `dotToUnderscore` util, replace all dots to underscore sign', () => {
      expect(dotToUnderscore(0.25)).toEqual('0_25');
    });
  });

  describe('getCssProperties', () => {
    it('renders default if state path not taken', () => {
      expect(getCssProperties({})).toEqual('');
    });
  });

  it('renders <Box /> component with flex properties', () => {
    const componentWithFlex: ShallowWrapper<BoxProps> = shallow(
      <Box flex={1}>{mockText}</Box>
    );

    expect(componentWithFlex.prop('style')).toEqual({
      flex: 1,
    });
    expect(componentWithFlex).toMatchSnapshot('flex');
  });

  it('renders <Box /> component with flex properties', () => {
    const componentWithFlex: ShallowWrapper<BoxProps> = shallow(
      <Box flex={{ grow: 1, shrink: 1, basis: '0%' }}>{mockText}</Box>
    );

    expect(componentWithFlex.prop('style')).toEqual({
      flexBasis: '0%',
      flexGrow: 1,
      flexShrink: 1,
    });
    expect(componentWithFlex).toMatchSnapshot('flex object');
  });

  it('renders <Box /> with hidden prop', () => {
    const component: ShallowWrapper<BoxProps> = shallow(
      <Box hidden>{mockText}</Box>
    );

    expect(component.hasClass('as-t-box--hidden')).toBe(true);
    expect(component).toMatchSnapshot('with hidden');
  });

  it('renders <Box /> with interactive prop', () => {
    const component: ShallowWrapper<BoxProps> = shallow(
      <Box interactive>{mockText}</Box>
    );

    expect(component.hasClass('as-t-box--interactive')).toBe(true);
  });

  it('renders <Box /> with tabIndex prop', () => {
    const component: ShallowWrapper<BoxProps> = shallow(
      <Box tabIndex={0}>{mockText}</Box>
    );

    expect(component.prop('tabIndex')).toBe(0);
  });

  it('renders <Box /> with role prop', () => {
    const component: ShallowWrapper<BoxProps> = shallow(
      <Box role="button">{mockText}</Box>
    );

    expect(component.prop('role')).toBe('button');
  });
});
