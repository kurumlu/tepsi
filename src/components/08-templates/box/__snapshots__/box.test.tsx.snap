// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders <Box /> component 1`] = `
<div
  className="as-t-box"
>
  That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.
</div>
`;

exports[`<Box /> renders <Box /> component with \`desktop\` and \`mobile\` properties: with props 1`] = `
<div
  className="as-t-box margin-left-mobile-1 margin-left-desktop-2"
>
  That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.
</div>
`;

exports[`<Box /> renders <Box /> component with flex properties: flex 1`] = `
<div
  className="as-t-box"
  style={
    Object {
      "flex": 1,
    }
  }
>
  That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.
</div>
`;

exports[`<Box /> renders <Box /> component with flex properties: flex object 1`] = `
<div
  className="as-t-box"
  style={
    Object {
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.
</div>
`;

exports[`<Box /> renders <Box /> with hidden prop: with hidden 1`] = `
<div
  className="as-t-box as-t-box--hidden"
>
  That’s why we pride ourselves on the fact that we love spending time outdoors, so when you ask us for advice, you know it comes from our real-world experiences.
</div>
`;
